lib/livebook_web/live/session_live/secrets_component.ex

```diff
@@ -71,16 +69,37 @@ defmodule LivebookWeb.SessionLive.SecretsComponent do
 
   @impl true
   def handle_event("save", %{"data" => data}, socket) do
-    secret = %{label: String.upcase(data["label"]), value: data["value"]}
-    Livebook.Session.put_secret(socket.assigns.session.pid, secret)
-    {:noreply, assign(socket, data: %{"label" => "", "value" => ""})}
+    if data_errors(data) == [] do
+      secret = %{label: String.upcase(data["label"]), value: data["value"]}
+      Livebook.Session.put_secret(socket.assigns.session.pid, secret)
+      {:noreply, push_patch(socket, to: socket.assigns.return_to)}
+    else
+      {:noreply, assign(socket, data: data)}
+    end
   end
 
   def handle_event("validate", %{"data" => data}, socket) do
     {:noreply, assign(socket, data: data)}
   end
 
-  defp data_valid?(data) do
-    String.match?(data["label"], ~r/^\w+$/) and data["value"] != ""
+  defp data_errors(data) do
+    Enum.flat_map(data, fn {key, value} ->
+      if error = data_error(key, value) do
+        [{String.to_atom(key), {error, []}}]
```

Comment:
Both `:label` and `:value` are used in this module

```suggestion
        [{String.to_existing_atom(key), {error, []}}]
```